src/basedtyping/transformer.py

```diff
@@ -0,0 +1,231 @@
+from __future__ import annotations
+
+import ast
+import types
+import typing
+import uuid
+from contextlib import contextmanager
+from enum import Enum
+from typing import cast
+
+import typing_extensions
+
+import basedtyping
+
+
+# ruff: noqa: N802, S101
+class CringeTransformer(ast.NodeTransformer):
+    """Transforms `1 | 2` into `Literal[1] | Literal[2]` etc"""
+
+    def __init__(
+        self,
+        globalns: dict[str, object] | None,
+        localns: dict[str, object] | None,
+        *,
+        string_literals: bool,
+    ):
+        self.string_literals = string_literals
+
+        # This logic for handling Nones is copied from typing.ForwardRef._evaluate
+        if globalns is None and localns is None:
+            globalns = localns = {}
+        elif globalns is None:
+            # apparently pyright doesn't infer this automatically
+            assert localns is not None
+            globalns = localns
+        elif localns is None:
+            # apparently pyright doesn't infer this automatically
+            assert globalns is not None
+            localns = globalns
+
+        self.typing_name = f"typing_extensions_{uuid.uuid4().hex}"
+        self.basedtyping_name = f"basedtyping_{uuid.uuid4().hex}"
```

Comment:
if this is dynamically adding random stuff to the global namespace they should be private
```suggestion
        self.typing_name = f"_typing_extensions_{uuid.uuid4().hex}"
        self.basedtyping_name = f"_basedtyping_{uuid.uuid4().hex}"
```

Reply:
i guess, but it wouldn't really make a difference to anything, it just gets evaluated as an implementation detail, noone would ever see it